File: API/WorkoutTracker.API/Models/EntityDTO.cs
using System;

namespace WorkoutTracker.API.Models
{
    /// <summary>
    /// A DTO representing an entity
    /// </summary>
    /// <param name="Id">The public ID of the entity. This is not the same as the internal, integer ID.</param>
    /// <param name="CreatedDateTime">The date/time when the entity was created.</param>
    /// <param name="ModifiedDateTime">The date/time when the entity was last modified, if applicable.</param>
    public abstract record EntityDTO(Guid Id, DateTime CreatedDateTime, DateTime? ModifiedDateTime);
}

File: workout-tracker/src/app/shared/models/entity-dto.ts
import { IMightHaveAuditDates } from "../interfaces/i-might-have-audit-dates";

export abstract class EntityDTO implements IMightHaveAuditDates {
  public id: string = ''; //GUID in API
  public createdDateTime: Date = new Date();
  public modifiedDateTime?: Date | null = null;
}